[Mervyn] ip

mervyn-teo opened the pull request:
### This is a Heading 😢😢😢

- This is a bullet list
- bullet
- buller

1. This is a numbered list
2. number 2
3. number 3

```java
 This is a code block 
```

Tasklist:
- [x] 123
- [ ] 345
- [ ] 567

> The quick brown fox jumps over the lazy dog

This is a link to [Google](https://www.google.com)

this is `inline` code

This is **BOLD**, _italics_, ~~strikethrough~~,

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mervyn-teo committed:
Made initial skeleton for chatbot.

mervyn-teo committed:
Made the echo functionality

mervyn-teo committed:
Added the ability to add, list and exit the program

commands:
- add: input by users will be automatically added to a list
- list: list all items in list
- bye: exits the program

mervyn-teo committed:
Added the ability to add, list and exit the program

commands:
- add: input by users will be automatically added to a list
- list: list all items in list
- bye: exits the program

mervyn-teo committed:
Added the ability to mark and unmark items in the list

commands added:
- mark [index]: mark the item with index number [index] as done
- unmark [index]: mark the item with index number [index] as not done

mervyn-teo committed:
Implemented Deadline, Event and ToDo functionalities. Made deadline, event and todo classes inherit from Task class for better organisation.

commands added:
- todo [item]: adds an todo item to the item list
- deadline [item] /by [date]: add a deadline item to the list
- event [item] /from [date] /to [date]: add an event item to the list

mervyn-teo committed:
added the functionality to show task counts whenever a task is added

mervyn-teo committed:
added tests

- input.txt: tests for basic input functionality
- dateTime.txt: test for commands involving dates
-  markTest.txt: test for commands involving mark & unmark

mervyn-teo committed:
Edited how spacers are printed, and changed text-ui-test accordingly

mervyn-teo committed:
Resolved some conflict and added errors.

mervyn-teo committed:
Added funtion to delete items in list

mervyn-teo committed:
Modified the code to use enums

mervyn-teo committed:
Skibidi: Add save/load functionality for task management

Task: Refactor to support polymorphism with JSON serialization

Task subtypes: Add default constructors and JSON annotations

   Build: Add Jackson library dependencies

   - Added `jackson-databind` and `jackson-annotations` to support JSON serialization/deserialization.

Build: Add Jackson library dependencies

GitIgnore: Add saved_list.json to ignored files

mervyn-teo committed:
Add saved_list.json to .gitignore

mervyn-teo committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Skibidi.java

mervyn-teo committed:
Implement saving and loading tasks for the delete command

mervyn-teo committed:
Merge branch 'refs/heads/add-gradle-support' into branch-Level-8

mervyn-teo committed:
Modify how date are recorded, change from String type to LocalDate type.

mervyn-teo committed:
Refactor code to make the code more OOP

mervyn-teo committed:
Remove redundant code

mervyn-teo committed:
Organise code into appropriate packages

mervyn-teo committed:
Correct test files

mervyn-teo committed:
ParserTest.java:
    - Add unit tests for parsing command strings in `Parser`.
CommandTest.java:
    - Add unit tests for processing commands in `Command`.

StorageTest.java:
    - Add unit tests for saving and loading tasks in `Storage`.

mervyn-teo committed:
Add javadocs for Command.java

mervyn-teo committed:
Add javadocs for Parser.java

mervyn-teo committed:
Add javadocs for Storage.java, Deadline.java, Event.java, Task.java, toDo.java

mervyn-teo committed:
Add javadocs for Messages.java, Skibidi.java, UI.java

mervyn-teo committed:
Modify code to comply with Google Java style guide

mervyn-teo committed:
Add functionality to find task

mervyn-teo committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/skibidi/command/Command.java
#	src/main/java/skibidi/ui/UI.java

mervyn-teo committed:
Merge branch 'refs/heads/branch-A-JavaDoc'

# Conflicts:
#	src/main/java/skibidi/command/Command.java
#	src/main/java/skibidi/command/Parser.java
#	src/main/java/skibidi/storage/Storage.java
#	src/main/java/skibidi/task/Task.java
#	src/main/java/skibidi/task/ToDo.java
#	src/main/java/skibidi/ui/UI.java

mervyn-teo committed:
Fix bug where delete always delete the first item

mervyn-teo committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Skibidi.java

alexandtheoh commented on build.gradle:
```diff
@@ -0,0 +1,48 @@
+}
+
+application {
+    mainClass.set("skibidi.ui.Skibidi")
```
Wow very skibidi!
You know I would be able to read your code alot better if you wrote it face to face instead of just handing me a friggin packet!

mervyn-teo replied:
Very **SIGMA** comment, I will take note of this in the future, keep up the skibidi!

alexandtheoh reviewed (commented):
Johnny walker in the war room? I studied...

mervyn-teo committed:
Implement auto code style checking using gradle's checkstyle.

flljy940 commented on src/main/java/skibidi/command/Command.java:
```diff
@@ -0,0 +1,318 @@
+            userChoiceEnum = Parser.parse(userChoice);
+
+            String[] splitUserchoice = userChoice.split(" ", 2);
+            switch (userChoiceEnum) {
```
LGTM. Just the indentation for switch-case needed to follow the coding standard.

flljy940 commented on src/main/java/skibidi/command/Command.java:
```diff
@@ -0,0 +1,318 @@
+        ui.printAdded(task.toString(), listItems.size());
+        storage.saveList(listItems);
+    }
+}
```
I like that everything is JavaDoc commented clearly.

flljy940 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,25 @@
+    ____________________________________________________________
+    ____________________________________________________________
+     skibidi bop bop!
+    ____________________________________________________________
```
Those tasks are lit!

mervyn-teo replied:
Thank you!🔥🔥🔥

flljy940 reviewed (commented):
Well done! Just a few nits to fix

mervyn-teo committed:
Fix all style issues within the code.

mervyn-teo committed:
Add GUI capabilities
Refactor Command.java to work with GUI

mervyn-teo committed:
Fix gradle configurations, modified Skibidi.java to work with current GUI

mervyn-teo committed:
Add assertion for Command.java

mervyn-teo committed:
Add assertion for Storage.java

mervyn-teo committed:
Replace printStackTrace methods with Loggers

Currently, PrintStackTrace is used for error handling

This is not robust enough as this does not enable logging of errors

Using Logger allow for more robust error handling as it is thread safe and user readable

Let's replace the use of printStackTrace to Loggers.

mervyn-teo committed:
Extract CommandType into a standalone file

CommandType is an Enum in the class Parser

The content of the CommandType is very bloated as it contains all commands

Let's extract CommandType into a standalone file allows for better readability of Parser class

mervyn-teo committed:
Merge pull request #1 from mervyn-teo/branch-A-Assertions

branch-A-Assertions

mervyn-teo committed:
Merge branch 'master' into branch-A-CodeQuality

mervyn-teo committed:
Merge pull request #2 from mervyn-teo/branch-A-CodeQuality

branch-A-CodeQuality

mervyn-teo committed:
Replace loop in findTask to streams

A for loop is used to cycle through existing items and find matching items

This is not very readable as it contains long logic that are difficult to keep up

 Let's use Streams to replace the for loop logic

 The use of stream shortens logic and make it readable

 The performance of streams are also optimised

mervyn-teo committed:
Merge branch 'master' into branch-A-Streams

mervyn-teo committed:
Merge pull request #3 from mervyn-teo/branch-A-Streams

branch-A-Streams

mervyn-teo committed:
Add tag functionality

mervyn-teo committed:
Add scrollwheel function

mervyn-teo committed:
Adjust GUI elements

mervyn-teo committed:
Remove dead code

mervyn-teo committed:
Rename title and add icon

mervyn-teo committed:
Update date format in chat

Update product page image

mervyn-teo committed:
Update code to save marked status

mervyn-teo committed:
Update README.md

mervyn-teo committed:
Update README.md

mervyn-teo committed:
Rename UI.png to Ui.png

mervyn-teo committed:
Update unit tests to function with current version of code
